src/lib/Components/Inbox/Conversations/Preview/InvoicePreview.tsx

```diff
@@ -64,14 +66,24 @@ const CostLabel = styled(PaymentRequest)`
 
 interface Props extends RelayProps {
   onSelected: () => void
+  conversationId: string
+  relay?: RelayRefetchProp
 }
 
 interface InvoiceStateButtonProps {
-  state: string | null
+  invoiceState: string | null
+  isOptimisticUpdate: boolean
 }
 
-const InvoiceStateButton: React.SFC<InvoiceStateButtonProps> = ({ state }) => {
-  switch (state) {
+const InvoiceStateButton: React.SFC<InvoiceStateButtonProps> = ({ invoiceState, isOptimisticUpdate }) => {
+  if (isOptimisticUpdate) {
```

Comment:
How about just passing `PAID` as the state from the call site when an optimisitic update is needed?